OloEngine/src/OloEngine/Scripting/Lua/LuaScriptGlue.cpp

```
@@ -47,10 +53,317 @@ namespace OloEngine
+        GetFn Get = nullptr;
+    };
+
+    // ── Safe proxy that re-resolves the component from EnTT on every access ──
+    //
+    // Lua never holds a raw T* into EnTT pool storage.  Instead get_component
+    // returns a LuaComponentProxy userdata.  __index / __newindex metamethods
+    // resolve Entity → Component& on every property read or write, so pool
+    // relocations (caused by AddComponent on other entities) cannot cause
+    // dangling-pointer UB.
+
+    struct LuaComponentProxy
```

Clang (14.0.6)

[warning] constructor does not initialize these fields: TypeName

(cppcoreguidelines-pro-type-member-init)